src/iuse_actor.cpp

```diff
@@ -1476,241 +1489,212 @@ int salvage_actor::time_to_cut_up( const item &it ) const
+    // Reinforcing does not decrease losses.
+    efficiency *= std::min( std::pow( 0.8, cut.get_item()->damage_level() ), 1.0 );
+
+    // Apply proportional item loss.
+    for( auto &iter : salvage ) {
+        iter.second *= efficiency;
     }
 
-    std::vector<item> stack{ *cut.get_item() }; /* working stack */
-    std::map<itype_id, int> salvage_to; /* outcome */
-    std::map<material_id, units::mass> mat_to_weight;
-    // Decompose the item into irreducible parts
-    while( !stack.empty() ) {
-        item temp = stack.back();
-        stack.pop_back();
+    auto distribute_uniformly = [&mat_to_weight]( item x, float efficiency ) -> void {
+        const float mat_total = std::max( x.type->mat_portion_total, 1 );
+        for( const auto &type : x.made_of() )
+        {
+            mat_to_weight[type.first] += x.weight() * ( static_cast<float>( type.second ) / mat_total ) *
+            efficiency;
+        }
+    };
+
+    // efficiency is decreased every time the ingredients of a recipe have more mass than the output
+    // num is passed along until we reach a basic component
+    std::function<void( item, int, float )> cut_up_component =
+        [&salvage, &mat_set, &distribute_uniformly, &cut_up_component]
+    ( item curr, int num, float eff_curr ) -> void {
 
         // If it is one of the basic components, add it into the list
-        if( temp.type->is_basic_component() ) {
-            salvage_to[temp.typeId()] ++;
-            continue;
+        if( curr.type->is_basic_component() )
+        {
+            int num_adjusted = static_cast<int>( num * eff_curr );
+            salvage[curr.typeId()] += num_adjusted;
+            return;
         }
+
         // Discard invalid component
-        std::set<material_id> mat_set;
-        for( std::pair<material_id, int> mat : cut_material_components ) {
-            mat_set.insert( mat.first );
-        }
-        if( !temp.made_of_any( mat_set ) ) {
-            continue;
-        }
-        //items count by charges should be even smaller than base materials
-        if( !temp.is_salvageable() || temp.count_by_charges() ) {
-            const float mat_total = temp.type->mat_portion_total == 0 ? 1 : temp.type->mat_portion_total;
-            // non-salvageable items but made of appropriate material, disrtibute uniformly in to all materials
-            for( const auto &type : temp.made_of() ) {
-                mat_to_weight[type.first] += ( temp.weight() * remaining_weight / temp.made_of().size() ) *
-                                             ( static_cast<float>( type.second ) / mat_total );
-            }
-            continue;
+        // Non-salvageable items are discarded even if made of appropriate material
+        // Items count by charges should be even smaller than base materials and therefore unsalvageable
+        if( !curr.made_of_any( mat_set ) || !curr.is_salvageable() || curr.count_by_charges() )
```

Comment:
I think this (specifically the `curr.count_by_charges()`) prevents one from cutting up bones into bone skewer. This used to work previously.

Reply:
Thanks, good to know because then the comment put there previously implied the wrong thing.